lrmd/tls_backend.c

```diff
@@ -260,6 +261,24 @@ lrmd_remote_listen(gpointer data)
         return TRUE;
     }
 
+#ifdef TCP_USER_TIMEOUT
+    {
+    int rc;
+    int optval;
+    long sbd_timeout = get_sbd_timeout();
+
+    if (sbd_timeout > 0) {
+        optval = sbd_timeout / 2; /* time to fail and retry before watchdog */
+        rc = setsockopt(csock, SOL_TCP, TCP_USER_TIMEOUT, &optval, sizeof(optval));
+        if (rc < 0) {
+            crm_perror(LOG_INFO, "writing TCP_USER_TIMEOUT (%d) on client socket failed", optval);
+            close(csock);
+            return TRUE;
+        }
+    }
+    }
+#endif
+
```

Comment:
check_sbd_timeout definitely belongs to lib/common/watchdog.c.

for functionalizing the code-block above I'd rather prefer lib/common/remote.c as it might
turn out to be a good thing to set the timeout for remote in general to make clients on 
remote-nodes more responsive. (driven by a cluster-property then so that the actual
tcp-user-timeout derives from sbd_timeout and e.g. remote-proxy-timeout)

Probably an overkill but ...: split off a generic crm_tcpconn_user_timeout_set in
lib/common/utils.c that could cover catching the possible unavailability of
the socket-option including a quiet-parameter (disables a log-warning
so that it can be used for more frequent occasions than excepting the
connection of a cluster-node at a remote-node).